consensus/consensus-types/src/common.rs

```diff
@@ -51,6 +51,36 @@ impl fmt::Display for TransactionSummary {
     }
 }
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Deserialize, Serialize, Hash, Ord, PartialOrd)]
+pub struct TransactionSynopsis {
```

Comment:
May be call it `TxnSummaryWithExpiration`?

Reply:
Actually instead of creating this new struct, does it make sense to add expiration time to the txn summary itself?

Reply:
Txn summary is used in a bunch of other places. Didn't want to change all of those for the sake of this small PR.
I can change `TxnSynopsis` to `TxnSummaryWithExpiration`.

Reply:
Hmm, I did take a look and it seems trivial to add `expiration_timestamp_secs` to the `TransactionSummary` itself.  What am I missing?

Reply:
There are 189 other occurrences of `TransactionSummary` in the code. Have to change how the transaction summary is calculated in each of those cases. Not sure if that is worth it.
<img width="584" alt="Screenshot 2024-07-01 at 2 35 35 PM" src="https://github.com/aptos-labs/aptos-core/assets/7092734/f5dab2a1-7eb1-45f2-b0f8-b52be280c1ff">

Reply:
You don't have to change all of the above places, only place you need to change is when you call the constructor - may be ~4 occurrences? 

<img width="1239" alt="Screenshot 2024-07-01 at 2 41 35 PM" src="https://github.com/aptos-labs/aptos-core/assets/11507549/8b6401ce-0501-49ca-a551-b58769ea7944">

Reply:
Yeah. You are right. 
I tried changing all those occurrences. As `TxnPtr` is an alias for `TransactionSummary`, tried changing those instantiations as well. Faced an issue when instantiating `TransactionSummary` from `OrderedQueueKey` where this expiration time is not present.